File: pythonsnippets/factorialFinder.py


limit = int(raw_input("N?   :"))
fact = 1
for num in range(1, limit):
    fact = fact * num
print fact

def factorialFinderRecurse():
    factFinderHelp(int(raw_input("N?    :")), 1)

def factFinderHelp(number, sum):
    if number == 0:
        print sum
    else:
        factFinderHelp(number - 1, sum * number)

factorialFinderRecurse()

File: rubysnippets/factorialFinder.rb



print "N?   :"
limit = gets.chomp.to_i
fact = 1
(1..limit).each do |i|
    fact = fact * i
end

puts "#{fact}\n"

def factorial_finder_recurse()
    print "N?    :"
    fact_finder_help(gets.chomp.to_i, 1)
end

def fact_finder_help(number, sum)
    if number == 0
        print sum
    else
        fact_finder_help(number - 1, sum * number)
    end
end

puts factorial_finder_recurse()
